modules/runner/src/integrationTest/java/org/apache/ignite/internal/runner/app/ItIgniteInMemoryNodeRestartTest.java

```diff
@@ -195,31 +200,29 @@ public void inMemoryNodeRestartNotLeader(TestInfo testInfo) throws Exception {
         InternalTableImpl internalTable = (InternalTableImpl) restartingTable.internalTable();
 
         // Check that it restarts.
-        assertTrue(waitForCondition(
-                () -> {
-                    boolean raftNodeStarted = loza.localNodes().stream().anyMatch(nodeId -> {
-                        if (nodeId.groupId() instanceof TablePartitionId) {
-                            return ((TablePartitionId) nodeId.groupId()).tableId() == table.tableId();
-                        }
+        waitForCondition(
+                () -> isRaftNodeStarted(table, loza) && assignmentsContain(restartingNodeConsistentId, internalTable),
+                TimeUnit.SECONDS.toMillis(10)
+        );
 
-                        return false;
-                    });
+        assertTrue(isRaftNodeStarted(table, loza), "Raft node is not started");
+        assertTrue(assignmentsContain(restartingNodeConsistentId, internalTable), "Assignments do not contain the node");
 
-                    if (!raftNodeStarted) {
-                        return false;
-                    }
+        // Check the data rebalanced correctly.
+        checkTableWithData(restartingNode, TABLE_NAME);
+    }
 
-                    Map<Integer, List<String>> assignments = internalTable.peersAndLearners();
+    private static boolean assignmentsContain(String restartingNodeConsistentId, InternalTableImpl internalTable) {
+        Map<Integer, List<String>> assignments = internalTable.peersAndLearners();
 
-                    List<String> partitionAssignments = assignments.get(0);
+        List<String> partitionAssignments = assignments.get(0);
```

Comment:
Why only the first partition?

Reply:
Because there is only one partition

Reply:
Maybe we can add a comment to the method or to the name of the method, which is only for the first and only partition?

Reply:
Renamed the method to indicate that it's checking the only partition that the table has